In [1]:
#mount Google drive on Colab
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [2]:
!ls '/content/drive/My Drive/Colab Notebooks'

FAO_Data.csv	Hamoye_intenship_StageA.ipynb  playground.ipynb
fuel_ferc1.txt	pchukwud-DA-Assignment2.ipynb


In [3]:
#import libraries needed for the project 
import pandas as pd
print('Successful!!!')

Successful!!!


> The Dataset was obtained from this link
https://github.com/HamoyeHQ/HDSC-Introduction-to-Python-for-machine-learning

In [4]:
#Read the comma-separated values (csv) file into DataFrame.
fao_data = pd.read_csv('/content/drive/My Drive/Colab Notebooks/FAO_Data.csv', encoding = 'latin-1')

In [5]:
#print the first five rows (observations) using head() function
fao_data.head()

Area Code     Area  Item Code         Item  Element Code  \
0          4  Algeria       2501   Population           511   
1          4  Algeria       2501   Population          5301   
2          4  Algeria       2901  Grand Total           664   
3          4  Algeria       2901  Grand Total           674   
4          4  Algeria       2901  Grand Total           684   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

>**Question 11:** What is the total sum of Animal Fat produced in 2014 and 2017 respectively?

In [6]:
#Perform a groupby sum aggregation on ‘Item’ and use the query function to isolate the Animal Fat Entry
fao_data.groupby('Item').sum().query('Item == "Animal fats"')

Area Code  Item Code  Element Code      Y2014      Y2015  \
Item                                                                    
Animal fats      85279    1882494       2567303  209460.54  200675.72   

                 Y2016      Y2017      Y2018  
Item                                          
Animal fats  183314.22  269617.53  269648.27

>>>> **Answer:** The total sum of Animal Fat produced in 2014 and 2017 are 209460.54 and 269617.53 tonnes respectively

**Question 12:** What is the mean and standard deviation across the whole dataset for the year 2015 to 3 decimal places?

In [7]:
#use the describe funtion to get the summary statistics of the entire dataset and ge the mean, std for the year 2015
fao_data.describe()['Y2015']

count     59395.000000
mean        135.235966
std        1603.403984
min       -3161.000000
25%           0.000000
50%           0.080000
75%           8.460000
max      181137.000000
Name: Y2015, dtype: float64

>>>> **Answer:** The mean and std across the dataset for the year 2015 is 135.236 and 1603.404 respectively

**Question 13:** What is the total number and percentage of missing data in 2016 to 2 decimal places?

In [8]:
#get the total number of entries using df.shape function or length of the dataframe
row, col = fao_data.shape
row

60943

In [9]:
#get the total number of missing values in the year 2016 
missing_data = fao_data.isnull().sum()['Y2016']
missing_data

1535

In [10]:
#calculate the percentage of missing data in 2016
percent = round((missing_data / row) * 100, 2)
percent

2.52

In [11]:
print('The total number and percentage of missing data in 2016 is {} and {}% respectively'.format(missing_data, percent))

The total number and percentage of missing data in 2016 is 1535 and 2.52% respectively


**Question 14:** Which year had the highest correlation with ‘Element Code’? 

In [12]:
#use the df.corr() function to get the correlation matrix of the entire dataset and observe the year with the highest coefficient with Element Code
fao_data.corr()

Area Code  Item Code  Element Code     Y2014     Y2015  \
Area Code      1.000000  -0.005159     -0.000209  0.006164  0.005472   
Item Code     -0.005159   1.000000     -0.024683  0.021722  0.020857   
Element Code  -0.000209  -0.024683      1.000000  0.024457  0.023889   
Y2014          0.006164   0.021722      0.024457  1.000000  0.994647   
Y2015          0.005472   0.020857      0.023889  0.994647  1.000000   
Y2016          0.005247   0.020109      0.023444  0.996081  0.995739   
Y2017          0.005006   0.021494      0.024254  0.995230  0.988048   
Y2018          0.005665   0.021314      0.024279  0.994872  0.988208   

                 Y2016     Y2017     Y2018  
Area Code     0.005247  0.005006  0.005665  
Item Code     0.020109  0.021494  0.021314  
Element Code  0.023444  0.024254  0.024279  
Y2014         0.996081  0.995230  0.994872  
Y2015         0.995739  0.988048  0.988208  
Y2016         1.000000  0.992785  0.992757  
Y2017         0.992785  1.000000  0.998103  
Y2018         0.992757  0.998103  1.000000

>>>> From the correlation matrix, 2014 has the highest correlation coefficient (0.024457) with Element Code

**Question 15:** What year has the highest sum of Import Quantity?

In [13]:
#Perform a groupby operation on ‘Element’ and use the resulting Dataframe to answer the question
fao_data.groupby('Element').sum().query('Element == "Import Quantity"')

Area Code  Item Code  Element Code      Y2014      Y2015  \
Element                                                                     
Import Quantity     688174   13795966      28834929  274144.48  267018.46   

                     Y2016      Y2017      Y2018  
Element                                           
Import Quantity  286582.78  294559.09  287997.09

>>>> 2017 has the highest sum of Import Quantity (294559.09) 

**Question 16:** What is the total number of the sum of Production in 2014?

In [14]:
#Perform a groupby operation on ‘Element’ and use the resulting Dataframe to answer the question
fao_data.groupby('Element').sum().query('Element == "Production"')

Area Code  Item Code  Element Code       Y2014       Y2015  \
Element                                                                  
Production     526751   10450053      21388191  1931287.75  1947019.39   

                 Y2016       Y2017       Y2018  
Element                                         
Production  1943537.15  2030056.89  2075072.89

>>>> The sum of production in 2014 is **1931287.75** tonnes

**Question 17:** Which of these elements had the highest sum in 2018?

In [15]:
#Perform groupby operation on 'Element', select 2018 and find the sum and sort the values in descending order
fao_data.groupby('Element')['Y2018'].sum().sort_values(ascending = False)

Element
Domestic supply quantity                  2161192.10
Production                                2075072.89
Food                                      1303841.28
Total Population - Both sexes             1140605.00
Food supply (kcal/capita/day)              455261.00
Processing                                 308429.00
Import Quantity                            287997.09
Feed                                       233489.68
Export Quantity                            181594.80
Losses                                     163902.00
Other uses (non-food)                       91300.97
Food supply quantity (kg/capita/yr)         49056.85
Residuals                                   34864.00
Seed                                        25263.14
Stock Variation                             20577.91
Protein supply quantity (g/capita/day)      11833.56
Fat supply quantity (g/capita/day)          10258.69
Tourist consumption                            90.00
Name: Y2018, dtype: float64

>>>> The element with highest sum in 2018 is Domestic Supply Quantity

**Question 18:** Which of these elements had the 3rd lowest sum in 2018?

In [16]:
#Perform groupby operation on 'Element', select 2018 and find the sum and sort the values in descending order
fao_data.groupby('Element')['Y2018'].sum().sort_values()

Element
Tourist consumption                            90.00
Fat supply quantity (g/capita/day)          10258.69
Protein supply quantity (g/capita/day)      11833.56
Stock Variation                             20577.91
Seed                                        25263.14
Residuals                                   34864.00
Food supply quantity (kg/capita/yr)         49056.85
Other uses (non-food)                       91300.97
Losses                                     163902.00
Export Quantity                            181594.80
Feed                                       233489.68
Import Quantity                            287997.09
Processing                                 308429.00
Food supply (kcal/capita/day)              455261.00
Total Population - Both sexes             1140605.00
Food                                      1303841.28
Production                                2075072.89
Domestic supply quantity                  2161192.10
Name: Y2018, dtype: float64

>>>> The element with the third lowest sum in 2018 is **Protein Supply Quantity**

**Question 19:** What is the total Import Quantity in Algeria in 2018?

In [17]:
#perform a groupby operation on the dataset on 'Element' and 'Area'. Sum the total of all the elements, filter out Algeria 
#import quantity from the resulting dataset
fao_data.groupby(['Element', 'Area']).sum().query('Area == "Algeria"').query('Element == "Import Quantity"')

Area Code  Item Code  Element Code     Y2014  \
Element         Area                                                    
Import Quantity Algeria        448     301030        628432  34940.12   

                            Y2015     Y2016     Y2017     Y2018  
Element         Area                                             
Import Quantity Algeria  37841.91  36736.08  36059.29  36238.29

>>>> The total Import Quantity in Algeria in 2018 is **36238.29**

**Question 20:** What is the total number of unique countries in the dataset?

In [18]:
#use the nunique() funtion on the column named Area
fao_data.Area.nunique()

49